src/core/modules/Landscape.cpp

```diff
@@ -156,32 +156,23 @@ void Landscape::loadCommon(const QSettings& landscapeIni, const QString& landsca
 
 	if (landscapeIni.childGroups().contains("location"))
 	{
-		if (landscapeIni.contains("location/planet"))
-			location.planetName = landscapeIni.value("location/planet").toString();
-		else
-			location.planetName = "Earth";
+		location.planetName = landscapeIni.value("location/planet", "Earth").toString();
 		// Tolerate decimal values in .ini file, but round to nearest integer
-		if (landscapeIni.contains("location/altitude"))
-			location.altitude = qRound(landscapeIni.value("location/altitude").toDouble());
-		if (landscapeIni.contains("location/latitude"))
-			location.setLatitude(static_cast<float>(StelUtils::getDecAngle(landscapeIni.value("location/latitude").toString())*M_180_PI));
-		if (landscapeIni.contains("location/longitude"))
-			location.setLongitude(static_cast<float>(StelUtils::getDecAngle(landscapeIni.value("location/longitude").toString())*M_180_PI));
+		location.altitude = qRound(landscapeIni.value("location/altitude", 0.).toDouble());
+		location.setLatitude(static_cast<float>(StelUtils::getDecAngle(landscapeIni.value("location/latitude", 0.).toString())*M_180_PI));
```

Comment:
Latitude can’t be outside the range -90..90 degrees, so, maybe using 100 degree as default values for wrong latitude?

Reply:
How will it behave though if applied?

Reply:
An invalid StelLocation should be just marked invalid by the dedicated flag, role='!', which I retry here and now. For now I added checks hopefully wherever StelLocations are generated, to set them valid only if useful. It may be useful to load other roles [CRNOL] from the landscape.ini. However, are these roles used anywhere in the program? Maybe to guesstimate light pollution?

The code change above changes nothing in behaviour BTW. long/lat was and still is 0 for a fresh StelLocation.  It seems in 2013 (!) we did not trust the default values and were super careful when parsing an ini file. If you think it's needed, we now can introduce qBound clamping in StelLocation::setLongitude() and StelLocation::setLatitude(). Or we just keep making the user responsible for own errors.